lib/rucio/core/replica.py

```diff
@@ -2456,15 +2620,19 @@ def get_and_lock_file_replicas(scope, name, nowait=False, restrict_rses=None, *,
     :returns:              List of SQLAlchemy Replica Objects
     """
 
-    query = session.query(models.RSEFileAssociation).filter_by(scope=scope, name=name).filter(models.RSEFileAssociation.state != ReplicaState.BEING_DELETED)
-    if restrict_rses is not None:
-        if len(restrict_rses) < 10:
-            rse_clause = []
-            for rse_id in restrict_rses:
-                rse_clause.append(models.RSEFileAssociation.rse_id == rse_id)
-            if rse_clause:
-                query = query.filter(or_(*rse_clause))
-    return query.with_for_update(nowait=nowait).all()
+    stmt = select(
+        models.RSEFileAssociation
+    ).where(
+        and_(models.RSEFileAssociation.scope == scope,
+             models.RSEFileAssociation.name == name,
+             models.RSEFileAssociation.state != ReplicaState.BEING_DELETED)
+    )
+    if restrict_rses is not None and len(restrict_rses) < 10:
+        rse_clause = [models.RSEFileAssociation.rse_id == rse_id for rse_id in restrict_rses]
+        stmt = stmt.where(or_(*rse_clause))
```

Comment:
I would still put this under `if rse_clause:` to keep it functionally the same as before